scripts/preprocess.py

```diff
@@ -0,0 +1,29 @@
+import os
+from argparse import ArgumentParser
+from shutil import copytree, rmtree
+
+
+def fix_naming(dataset_path: str):
+    train = dataset_path + "java-med-psi.train.c2s"
+    val = dataset_path + "java-med-psi.val.c2s"
+    os.remove(train)
+    os.remove(val)
+
+
+def preprocess(project_path: str):
+    project_name = os.path.basename(os.path.normpath(project_path))
+    dataset_path = "../datasets/" + project_name + "/java-med-psi/"
+    new_path = "temporary/test/" + project_name
+    copytree(project_path, new_path)
+    cmd = 'bash ./psiminer/psiminer.sh "{}" "{}" psiminer_config.json'.format("./temporary", dataset_path)
+    os.system(cmd)
+    rmtree("temporary")
```

Comment:
For temporary directories, you can use built-in Python logic. See [here](https://docs.python.org/3/library/tempfile.html) and [here](https://stackoverflow.com/questions/3223604/how-to-create-a-temporary-directory-and-get-its-path-file-name)